Point_set_processing_3/include/CGAL/scanline_orient_normals.h

```diff
@@ -0,0 +1,552 @@
+     range is cutted everytime the flag (which tells if the scanner
+     was moving in the positive or negative direction) changes.
+
+   - if no direction flag map is provided, a fallback method simply
+     cuts the range everytime 3 consecutive points form an acute angle
+     on the projected XY-plane. This fallback method gives suboptimal
+     results.
+
+   Then, the line of sight (estimated vector between a point and the
+   position of the scanner at its time of acquisition) is estimated:
+
+   - if `scan_angle` is provided, the line of sight can be directly
+     computed as a combination of the estimated scanline and of the
+     scan angle.
+
+   - if no scan angle map is provided, then for each scanline, the
+     position of the scanner is estimated as being above of the
+     barycenter of the points of the scanline projected on the
+     XY-plane. This fallback method gives suboptimal results.
+
+   Once the line of sight is estimated for each point, the normals are
+   oriented by checking if, for one each, the line of sight and the
+   normal vector give a positive scalar product. If they don't, then
+   the normal vector is inverted.
+
+   \note this method gives optimal results when `scan_direction_flag`
+   and `scan_angle` are provided. Correct results may still be
+   produced in the absence of either one or both of these properties,
+   as long as the point set is ordered in 2.5D scanlines.
+
+   \tparam PointRange is a model of `Range`. The value type of
+   its iterator is the key type of the named parameter `point_map`.
+
+   \param points input point range.
+   \param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+
+   \cgalNamedParamsBegin
+     \cgalParamNBegin{point_map}
+       \cgalParamDescription{a property map associating points to the elements of the point set `points`}
+       \cgalParamType{a model of `ReadablePropertyMap` whose key type is the value type
+                      of the iterator of `PointRange` and whose value type is `geom_traits::Point_3`}
+       \cgalParamDefault{`CGAL::Identity_property_map<geom_traits::Point_3>`}
+     \cgalParamNEnd
+
+     \cgalParamNBegin{normal_map}
+       \cgalParamDescription{a property map associating normals to the elements of the point set `points`}
+       \cgalParamType{a model of `WritablePropertyMap` whose key type is the value type
+                      of the iterator of `PointRange` and whose value type is `geom_traits::Vector_3`}
+     \cgalParamNEnd
+
+     \cgalParamNBegin{scan_angle_map}
+       \cgalParamDescription{a property map associating the angle of acquisition to the elements of the point set `points`}
+       \cgalParamType{a model of `ReadablePropertyMap` whose key type is the value type
+                      of the iterator of `PointRange` and whose value type is `float`}
+     \cgalParamNEnd
+
+     \cgalParamNBegin{scan_direction_flag}
```

Comment:
Where does `flag` comes from? As a user I'm not sure how to use it. Can it be any value as long as I use identical values to partition the input range?

Reply:
From the LAS standard again. It's true that I should document the values. This could actually be a boolean map (values are 1 and 0). In practice, your assumption (using any value as long as it partition the range) is correct, so I should maybe change the name and description to match this.